src/bin/pg_autoctl/pgsql.c

```diff
@@ -470,28 +504,11 @@ pgsql_retry_open_connection(PGSQL *pgsql, uint64_t startTime)
 	while (!connectionOk)
 	{
 		int sleep = 0;
-		uint64_t now = time(NULL);
 
-		/* Any signal is reason enough to break out from this retry loop. */
-		if (asked_to_quit ||
-			asked_to_stop ||
-			asked_to_stop_fast ||
-			asked_to_reload)
+		if (pgsql_retry_policy_expired(&(pgsql->retryPolicy)))
```

Comment:
Afaict this call does something completely different than the code that was here before.
1. It doesn't break when it reaches a signal, instead it enters this if block.
2. It does the opposite of what it previously did whit `true`/`false` for the `maxR`+`maxT`.

Reply:
I think your later comment is right, the return values have been inverted in that function.

Reply:
The first item still stands. With the new code it will log the connection error on any signal, with the old code it wouldn't. I'm not sure if we care about that though.

Reply:
Yeah I think it's better to log about “cancelling” the connection attempt, it might be what the user is trying to achieve with interrupting (C-c) the flow at this point.